typescript/src/wallet.ts

```diff
@@ -0,0 +1,98 @@
+import { BigNumber } from "ethers"
+import { Hex } from "./hex"
+
+export enum WalletState {
+  /**
+   * The wallet is unknown to the Bridge.
+   */
+  Unknown = 0,
+  /**
+   * The wallet can sweep deposits and accept redemption requests.
+   */
+  Live = 1,
+  /**
+   * The wallet was deemed unhealthy and is expected to move their outstanding
+   * funds to another wallet. The wallet can still fulfill their pending redemption
+   * requests although new redemption requests and new deposit reveals are not
+   * accepted.
+   */
+  MovingFunds = 2,
+  /**
+   *  The wallet moved or redeemed all their funds and is in the
+   * losing period where it is still a subject of fraud challenges
+   * and must defend against them.
+   *  */
+  Closing = 3,
+  /**
+   * The wallet finalized the closing period successfully and can no longer perform
+   * any action in the Bridge.
+   * */
+  Closed = 4,
+  /**
+   * The wallet committed a fraud that was reported, did not move funds to
+   * another wallet before a timeout, or did not sweep funds moved to if from
+   * another wallet before a timeout. The wallet is blocked and can not perform
+   * any actions in the Bridge.
+   */
+  Terminated = 5,
+}
+
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace WalletState {
+  export function parse(val: number): WalletState {
+    return (
+      (<never>WalletState)[
+        Object.keys(WalletState)[
+          Object.values(WalletState).indexOf(val as WalletState)
+        ]
+      ] ?? WalletState.Unknown
+    )
+  }
+}
+
+/**
+ * Represents a deposit.
+ */
+export interface Wallet {
+  /**
+   * Identifier of a ECDSA Wallet registered in the ECDSA Wallet Registry.
+   */
+  ecdsaWalletID: string
+  /**
+   * Compressed public key of the ECDSA Wallet.
+   */
+  walletPublicKey: Hex
+  /**
+   * Latest wallet's main UTXO hash.
+   */
+  mainUtxoHash: string
```

Comment:
We can use `Hex` here.

Reply:
It's not `TransactionHash`, right?

Reply:
Yeah, it's not.

Reply:
c83a463a81160e27fd4eb4d3763f71a2ee3a989a